nanoFramework.Hardware.Esp32.Rmt/RmtChannelSettings.cs

```diff
@@ -0,0 +1,130 @@
+        /// </summary>
+        /// <remarks>
+        /// This function is used to configure the amount of memory blocks allocated to a channel.
+        /// The 8 channels share a 512x32-bit RAM block which can be read and written by the processor cores over the APB bus,
+        /// as well as read by the transmitters and written by the receivers.
+        /// This means that every channel gets a single memory block equaling 64x32-bit.
+        /// If memory block number of one channel is set to a value greater than 1, this channel will occupy the memory block of the next channel.
+        /// Channel 0 can use at most 8 blocks of memory, accordingly channel 7 can only use one memory block.
+        /// </remarks>
+        public byte NumberOfMemoryBlocks { get => _numberOfMemoryBlocks; set => _numberOfMemoryBlocks = value; }
+
+        /// <summary>
+        /// Gets or sets the RMT Ring Buffer size.
+        /// </summary>
+        /// <remarks>
+        /// The ring buffer is available in transmit and receive channels.
+        /// For receive channels, Incoming <see cref="RmtCommand"/>s are moved to the ring buffer after the <see cref="ReceiverChannelSettings.IdleThreshold"/> has lapsed.
+        /// For transmit channels, <see cref="RmtCommand"/>s are automatically copied over to the channel's memory block and written to the transmitter.
+        /// </remarks>
+        public int BufferSize { get => _bufferSize; set => _bufferSize = value; }
+
+        /// <summary>
+        /// Gets or sets a value indicating if the RMT module should invert the incoming/outgoing signal.
+        /// </summary>
+        /// <remarks>This works like an external inverter connected to the GPIO of certain RMT channel.</remarks>
+        public bool SignalInverterEnabled { get => _signalInverterEnabled; set => _signalInverterEnabled = value; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RmtChannelSettings"/> class.
+        /// </summary>
+        /// <param name="pinNumber">The GPIO Pin number to use with the channel.</param>
+        protected RmtChannelSettings(int pinNumber)
+            : this(channel: -1, pinNumber)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RmtChannelSettings"/> class.
+        /// </summary>
+        /// <param name="channel">The channel number to use. Valid value range is 0 to 7 (inclusive).</param>
+        /// <param name="pinNumber">The GPIO Pin number to use with the channel.</param>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="channel"/> must be between 0 and 7.</exception>
+        protected RmtChannelSettings(int channel, int pinNumber)
+        {
+            this.Channel = channel <= 7
+                ? channel
+                : throw new ArgumentOutOfRangeException(nameof(channel));
+
+            this.PinNumber = pinNumber;
```

Comment:
Better don't use the `this.` accessor. Go straight to the backing field.
Same for the one after this and the next constructor too.